src/plugins/intel_npu/src/plugin/npuw/weights_bank.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <mutex>
+#include <unordered_map>
+
+#include "openvino/openvino.hpp"
+#include "openvino/runtime/intel_npu/level_zero/level_zero.hpp"
+#include "openvino/runtime/iremote_context.hpp"
+#include "openvino/runtime/itensor.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "plugin.hpp"
+
+namespace ov {
+namespace npuw {
+namespace weights_bank {
+
+class WeightBank {
+public:
+    WeightBank(const std::string& device_name, const std::shared_ptr<const ov::IPlugin>& plugin)
```

Comment:
What's the reason to have both device name and the plugin?

Reply:
Due to NPU specific fields initialization

Reply:
Lets drop it for now

Reply:
Done